src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -421,13 +444,20 @@ private void Insert(TKey key, TValue value, bool add)
             {
                 if (entries[i].hashCode == hashCode && comparer.Equals(entries[i].key, key))
                 {
-                    if (add)
+                    switch (behavior)
```

Comment:
`switch`? I wouldn't do that. If the C# and JIT compilers decide to generate a jump table based switch (and it looks like they do that unfortunately) both perf and code size will suffer.

Besides, the code ends up being rather convoluted. The None case breaks to return false while the OverwriteExisting case returns true directly. And ThrowOnExisting breaks but of course it doesn't really do that in reality.

Just use `if`
```C#
if (behavior == InsertionBehavior.ThrowOnExisting)
    ThrowHelper.ThrowAddingDuplicateWithKeyArgumentException(key); 
if (behavior == InsertionBehavior.None)
    return false;
Debug.Assert(behavior == InsertBehavior.OverwriteExisting); // if you want
entries[i].value = value; 
version++; 
return true; 
```